src/XGBoostSharp/lib/DMatrix.cs

```diff
@@ -7,7 +7,7 @@ namespace XGBoostSharp.lib;
 public class DMatrix : IDisposable
 {
     readonly SafeDMatrixHandle m_safeDMatrixHandle;
-    readonly float m_missing = -1.0F; // arbitrary value used to represent a missing value
+    const float DefaultMissing = float.NaN;
```

Comment:
this breaks backwards compatibility, so I think if this changed is merged, you'd have to release in a new major version

Reply:
@Nasicus We are still at major version 0, so anything MAY still change at any time :). Once major version 1 has been released, you would be right of course. But since the API is not stable yet we should stick with major version 0 for a while still.